### ISSUE TITLE
Incorrect output of Record with numeric key

### ISSUE THREAD

**Issue by matt-pawley** [vega/ts-json-schema-generator#1262]
A `Record` with a numeric key is incorrectly outputted as an `array` instead of an `object`. 

```typescript
// foo.ts

interface RecordValue {
  baz: 'baz';
}

export type Foo = Record<number, RecordValue>;

// Note, the following output as expected:

// export type Foo = Record<string, RecordValue>;

// export type Foo = {
//   [key: number]: RecordValue;
// };
```

Command:
```
ts-json-schema-generator --path 'foo.ts'
```

Actual Result:
```
{
  "$ref": "#/definitions/Foo",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Foo": {
      "items": {
        "additionalProperties": false,
        "properties": {
          "baz": {
            "const": "baz",
            "type": "string"
          }
        },
        "required": [
          "baz"
        ],
        "type": "object"
      },
      "type": "array"
    }
  }
}
```

Expected Result:
```
{
  "$ref": "#/definitions/Foo",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Foo": {
      "additionalProperties": {
        "additionalProperties": false,
        "properties": {
          "baz": {
            "const": "baz",
            "type": "string"
          }
        },
        "required": [
          "baz"
        ],
        "type": "object"
      },
      "type": "object"
    }
  }
}
```

Fully appreciate all object keys are strings in Javascript, so declaring a Record with a numeric interface is unusual. However. typescript allows it, so would be useful for `ts-json-schema-generator` to also support it.

**Comment by matt-pawley** [comment]
Found the issue is in `MappedTypeNodeParser` - https://github.com/vega/ts-json-schema-generator/blob/next/src/NodeParser/MappedTypeNodeParser.ts#L57-L59

If we move `keyListType instanceof NumberType` into the above `if` statement, it works as expected. However this breaks https://github.com/vega/ts-json-schema-generator/blob/next/test/valid-data/type-mapped-array/main.ts

**Comment by matt-pawley** [comment]
Fixed in [v1.1.0](https://github.com/vega/ts-json-schema-generator/releases/tag/v1.1.0)
